build-script.py

```diff
@@ -517,26 +504,37 @@ def verify_code_generated_files(
     user_swiftideutils_generated_dir = os.path.join(
         IDEUTILS_DIR, "generated"
     )
+    user_swiftparser_generated_dir = os.path.join(
+        SWIFTPARSER_DIR, "generated"
+    )
     user_swiftsyntax_generated_dir = os.path.join(
         SWIFTSYNTAX_DIR, "generated"
     )
     user_swiftsyntaxbuilder_generated_dir = os.path.join(
         SWIFTSYNTAXBUILDER_DIR, "generated"
     )
 
-    self_swiftbasicformat_generated_dir = tempfile.mkdtemp()
-    self_swiftideutils_generated_dir = tempfile.mkdtemp()
-    self_swiftparser_generated_dir = tempfile.mkdtemp()
-    self_swiftsyntax_generated_dir = tempfile.mkdtemp()
-    self_swiftsyntaxbuilder_generated_dir = tempfile.mkdtemp()
+    self_temp_dir = tempfile.mkdtemp()
+
+    self_swiftbasicformat_generated_dir = os.path.join(
+        self_temp_dir, "Sources", "SwiftBasicFormat", "generated"
+    )
+    self_swiftideutils_generated_dir = os.path.join(
+        self_temp_dir, "Sources", "IDEUtils", "generated"
+    )
+    self_swiftparser_generated_dir = os.path.join(
+        self_temp_dir, "Sources", "SwiftParser", "generated"
+    )
+    self_swiftsyntax_generated_dir = os.path.join(
+        self_temp_dir, "Sources", "SwiftSyntax", "generated"
+    )
+    self_swiftsyntaxbuilder_generated_dir = os.path.join(
+        self_temp_dir, "Sources", "SwiftSyntaxBuilder", "generated"
+    )
```

Comment:
Since e you are refactoring this: I think this can become a loop now.

```python
self_temp_dir = tempfile.mkdtemp()

run_code_generation(…)

for module in ["SwiftBasicFormat", "IDEUtils", …]:
  self_generated_dir = os.path.join(self_temp_dir, "Sources", module, "generated")
  user_generated_dir = os.path.join(<something>, module, "generated")
  check_generated_files_match(self_generated_dir, user_generated_dir)
```